app/controllers/routes_controller.rb

```diff
@@ -53,4 +53,17 @@ def valid_dates?
   def are_dates_present?
     params[:init_date].present? && params[:end_date].present?
   end
+
+  def route_add_length_and_points
+    route.update(length: params[:length])
+    params['points'].map { |coord| Location.create(route_id: route.id, latitude: coord.first, longitude: coord.last) }
```

Comment:
we shouldn't create the locations if the route cannot be updated. So maybe we can move this to a method an call it after we are sure the route was updated.